### ISSUE TITLE
Missing code for DWEN

### ISSUE THREAD

**Issue by happygirlzt** [irving-muller/soft_alignment_model_bug_deduplication#5]
Hi @irving-muller ! (Sorry.. I'm here again 😅)
I found in your paper, you compared with DWEN (Poster: DWEN: Deep Word Embedding Network for Duplicate Bug Report Detection in Software Repositories. [ICSE'18-Companion])
However, I cannot find the code for it. ( I guess you missed including it 😸 ) I know this is not the focus of your work, but I just wonder whether it is possible to add these scripts to this repo. 😃  Thanks anyway for your consistent patience and genuine help.
BTW, I found you actually included DBR-CNN code both under `model` and `experiments` folders, while in your paper, you did not include the results from them 😺 )
Another question is: I have no idea what is `experiments/compare_aggregate_model.py` for. Does it seem you did not use it for the final use? Or it is for the implementation of `REP` (Towards More Accurate Retrieval of Duplicate Bug Reports.)[ASE11]
Thank you so much again and have a nice day 😆

**Comment by irving-muller** [comment]
Hello @happygirlzt 

No worries!! :)

I used the same implementation of Siamese Pair to implement DWEN. To run DWEN on a specific dataset, you have to pass the correct JSON configuration (`json_parameters/dwen_test_dataset.json`) to `experiments/siamese_pairs.py`. For instance, the following command executes DWEN on Eclipse: `python experiments/siamese_pairs.py --name "dwen_test_mozilla" -F HOME_DIR/experiments with HOME_DIR/son_parameters/dwen_test_mozilla.json "recall_rate.window=365"`. Sorry this is not clear on the README.

We have implemented DBR-CNN but, unfortunately, the results were really low. We did not know if the bad performance was related to our implementation or the method and we decided to not include them in the paper. Moreover, since DBR-CNN is related to siamese pair, we believe that the comparison is not necessary.

`experiments/compare_aggregate_model.py` should have been removed from the repository. It is the first version of SABD. 

Please let me know if you have any question

**Comment by happygirlzt** [comment]
Dear @irving-muller , I really appreciate your patience and help! 😃  I understand it better now! 👍 

Regarding the json-parameters files, I have two questions. ① I found the two json-parameter files`json_parameters/siamese_pair_test_open_office.json` and `json_parameters/siamese_pairs_test_open_office.json`. I figured out there are just two parts are different: (1) word embeddings (2) batch_normalization. Which one should I use? I used `json_parameters/siamese_pair_test_open_office.json` and it worked fine.
<img width="1424" alt="Screenshot 2021-06-16 at 13 47 00" src="https://user-images.githubusercontent.com/15061984/122164389-55403c00-cea9-11eb-8f68-ead3aa677c4d.png">
② Which model is the `json_parameters/fast_cadd_open_office_test.json` for?

BWT, For **REP** and **BM25_ext**, I should refer to [fast-dbrd-modified](https://github.com/irving-muller/fast-dbrd-modified), right? They are not included in this repo.

Many thanks for your kindness! :) Have a nice day! ❤️

**Comment by irving-muller** [comment]
Hello @happygirlzt . Sorry for the delay.

For the experiments in the paper, I used the second one. The only difference between them is the batch normalization which, as far I remember, it did not significantly change the results. *.npy and *.lxc is faster to read from the disk than *.txt but, in the end, they contain the same word and embeddings.

> ② Which model is the json_parameters/fast_cadd_open_office_test.json for?

Please ignore `json_parameters/fast_cadd_open_office_test.json`. It should have been removed from the repository.

> BWT, For REP and BM25_ext, I should refer to fast-dbrd-modified, right? They are not included in this repo.

Yes, that is correct.

No worries. Have a nice day. :)

**Comment by happygirlzt** [comment]
Nice! Many thanks again for your detailed reply. ❤️ 
Have a great Friday and weekend!
